include/app-framework/QueueRegistry.icc

```diff
@@ -21,27 +22,29 @@ QueueRegistry::get_queue(std::string name) {
 
         throw QueueTypeMismatch(ERS_HERE, name, realname_source, realname_target);
       }
+
+      return queuePtr;
   } 
 
   auto itP = this->queue_configmap_.find(name);
   if ( itP != queue_configmap_.end() ) {
     QueueEntry entry = {&typeid(T), create_queue<T>(name, itP->second)};
     queue_registry_[name] = entry;
-    return std::dynamic_pointer_cast<NamedQueueI<T>>(entry.instance);
+    return std::dynamic_pointer_cast<QueueI<T>>(entry.instance);
 
   } else {
     throw std::runtime_error("Queue not found");
   }
 }
 
-
 template<typename T>
 std::shared_ptr<NamedObject> QueueRegistry::create_queue(std::string name, const QueueConfig& config) {
 
   std::shared_ptr<NamedObject> queue;
   switch(config.kind) {
     case QueueConfig::std_deque:
-      queue = std::make_shared<NamedStdDeQueue<T>>(name, config.size);
+      queue = makeNamedObject(GetClassName<T>("StdDeQueue"), name);
```

Comment:
Using this method to allocate queue is useless if we are bound by hard-coded type with a switch. I'll use this mechanism only for DAQModules